MAINTENANCE.md

```diff
@@ -0,0 +1,89 @@
+# SDK GO Maintenance 
```

Comment:
I have dediced to create dedicated `Maintenance.md` as new standard for our upstream repositories. 
This will be different audience than users reading https://github.com/mongodb/atlas-sdk-go/blob/main/CONTRIBUTING.md. 

Contributing.md is usually dedicated to the developers of the SDK (team and community) where maintenance is more support and help oriented with more insights into how things are automated, released etc.